lib/stages/visAccumulate.cpp

```diff
@@ -431,32 +426,48 @@ void visAccumulate::main_thread() {
 }
 
 
-void visAccumulate::initialise_output(visAccumulate::internalState& state, int in_frame_id,
-                                      int freq_ind) {
+bool visAccumulate::initialise_output(visAccumulate::internalState& state, int in_frame_id) {
+    // TODO: CHIME
+    auto metadata = (const chimeMetadata*)in_buf->metadata[in_frame_id]->metadata;
 
-    allocate_new_metadata_object(state.buf, state.frame_id + freq_ind);
-    auto frame = visFrameView(state.buf, state.frame_id + freq_ind, num_elements, 0);
+    for (size_t freq_ind = 0; freq_ind < num_freq_in_frame; freq_ind++) {
 
-    // Copy over the metadata
-    // TODO: CHIME
-    frame.fill_chime_metadata((const chimeMetadata*)in_buf->metadata[in_frame_id]->metadata);
+        if (wait_for_empty_frame(state.buf, unique_name.c_str(), state.frame_id + freq_ind)
+            == nullptr) {
+            return true;
+        }
+
+        allocate_new_metadata_object(state.buf, state.frame_id + freq_ind);
+        state.frames.emplace_back(state.buf, state.frame_id + freq_ind, num_elements, 0);
+        auto& frame = state.frames[freq_ind];
+
+        // Copy over the metadata
+        // TODO: CHIME
+        frame.fill_chime_metadata(metadata);
 
-    // TODO: set frequency id in some sensible generic manner
-    frame.freq_id += freq_ind;
+        // TODO: set frequency id in some sensible generic manner
+        frame.freq_id += freq_ind;
```

Comment:
I think this line will cause problems at the moment.   In the 16-element case I had to hardcoded this to `bin_number_16_elem(stream_id)` if `num_freq_in_frame == 128` and `bin_number_chime(stread_id)` if `num_freq_in_frame == 1` since I never got around to doing the `sensible generic` thing.   It looks like the `fill_chime_metadata()` function is setting `frame.freq_id = bin_number_chime(stream_id)`, which does work in the CHIME case, but then this multi-freq code will just overwrite it with `0`.

Reply:
I don't quite understand your comment here. It's clear that it's not going to label things correctly in the multifrequency case, which I think is fine, some of the CHIME specific stuff just needs replacing to make that work, and I think we both agree on that.

This does work fine for CHIME though, which you seem to suggest it won't. It sets the id in `fill_chime_metadata`, and then *increments* it by zero in the line below.

Reply:
Ah yes it's `+=` not `=`, so that does work; sorry I misread that.  And yes both agree we need to replace the CHIME specific streamID mapping to make this work in the multi-frequency case.

Reply:
Although for any system involving an ICEBoard we won't have sequential frequencies.  So when we fix this, it will need to be something which looks up the frequency in some way here.  A simple `+= counter` won't work in any of our existing multi-frequency setups.

Reply:
Kudos for attempting to read it well enough that you flagged up a small detail like this! :)